lib/cog/chat/slack/connector.ex

```diff
@@ -323,6 +332,44 @@ defmodule Cog.Chat.Slack.Connector do
     end
   end
 
+  defp normalize_channel_id(:by_id, id = "C" <> _, slack) do
+    case slack.channels[id] do
+      nil -> classify_error(id)
+      _channel -> {:channel, id}
+    end
+  end
+  defp normalize_channel_id(:by_name, name, slack) do
+    # Private groups and channels are both noted with the '#' but they are
+    # tracked separately. So we have to merge the two lists before searching.
+    channels = slack.channels
```

Comment:
Good point, originally I wasn't doing the group lookup bit, I was just looking at the channel. I'll rename the var. Honestly I'm not super happy with the name of the function either, but I wasn't sure what else to call it. We actually verify that we have an id, if we don't we assume it's the name of a channel or user and then do a lookup.